rosdep/python.yaml

```diff
@@ -7267,6 +7267,14 @@ python3-pymongo:
   nixos: [python3Packages.pymongo]
   openembedded: [python3-pymongo@meta-python]
   ubuntu: [python3-pymongo]
+python3-pynmea2:
+  debian:
+    pip:
+      packages: [pynmea2]
+  fedora:
+    pip:
+      packages: [pynmea2]
```

Comment:
New keys should not mix pip and system packages.

Please only add definitions for system packages in non-pip keys.

Reply:
I am a little confused by this. Based on the above comment Debian can be moved to a system package but still leaves Fedora using pip.

Would adding a new package such as `python3-pynmea2-pip` that addresses Fedora work this way there isn't a mixture of pip and system packages?

Reply:
I decided to add the `-pip` key to cover fedora. Pushing on next commit.

Reply:
@nuclearsandwich We can allow pip to backfill unsupported platforms and versions for the regular key that we believe will never have conflicting native packages. Splitting the key into two like this means that you can't use the same dependencies in the package.xml on two platforms. The `-pip` suffix is to distinguish keys which are pip only and on no other installer.

Reply:
I reverted the last commit and updated the single key to support both pip and system packages.